op-deployer/pkg/deployer/state/intent.go

```diff
@@ -32,35 +33,313 @@ func (d DeploymentStrategy) Check() error {
 	}
 }
 
+type IntentConfigType string
+
+const (
+	IntentConfigTypeTest              IntentConfigType = "test"
+	IntentConfigTypeStandard          IntentConfigType = "standard"
+	IntentConfigTypeCustom            IntentConfigType = "custom"
+	IntentConfigTypeStrict            IntentConfigType = "strict"
+	IntentConfigTypeStandardOverrides IntentConfigType = "standard-overrides"
+	IntentConfigTypeStrictOverrides   IntentConfigType = "strict-overrides"
+)
+
 var emptyAddress common.Address
+var emptyHash common.Hash
 
 type Intent struct {
-	DeploymentStrategy DeploymentStrategy `json:"deploymentStrategy" toml:"deploymentStrategy"`
+	DeploymentStrategy    DeploymentStrategy `json:"deploymentStrategy" toml:"deploymentStrategy"`
+	IntentConfigType      IntentConfigType   `json:"intentConfigType" toml:"intentConfigType"`
```

Comment:
[nit] can we make this just `configType`? `intent.intentConfigType` is a stutter due to the double `intent`.